### PR TITLE
add signal & fork function etc.

### PR DESCRIPTION
**Pull request by xiaobiaozhao**
add
1. add signal * function and other related function
2. add fork function
3. add wait & waitpid function

### CODE REVIEW

**Comment by doujiang24** [comment]
@spacewander @rainingmaster  Will you help to take a look at this?

File: lib/resty/signal.lua [hunk 1]
```diff
@@ -73,6 +76,59 @@ if not pcall(function () return C.strerror end) then
...
+    ]]
+end
+
+
+if not pcall(function() return C.signal end) then
+    ffi.cdef [[
+        typedef void (*sighandler_t)(int);
+        sighandler_t signal(int signum, sighandler_t handler);
+    ]]
+end
```
Comment (doujiang24) [inline lib/resty/signal.lua:92]: duplicated lines?

Comment (xiaobiaozhao) [reply lib/resty/signal.lua]: I had deleted duplicated lines

File: lib/resty/signal.lua [hunk 2]
```diff
@@ -151,6 +206,153 @@ function _M.kill(pid, sig)
...
+    return C.sigfillset(sigset)
+end
+
+
+local function sigaddset(sigset, name)
+    if not sigset then
+        return nil, "sigset is nil"
+    end
+    
+
```
Comment (doujiang24) [inline lib/resty/signal.lua:239]: style: duplicated empty lines.

**Comment by agentzh** [comment]
nginx's fds and other resources may leak into the child processes created by `fork()`. This PR doesn't handle such details.

Also I wonder what's your use cases for the `fork()` call here in the OpenResty context?

**Comment by xiaobiaozhao** [comment]
> nginx's fds and other resources may leak into the child processes created by `fork()`. This PR doesn't handle such details.
> 
> Also I wonder what's your use cases for the `fork()` call here in the OpenResty context?

image

**Pull request closed without merging**